tests/contrib/executors/test_kubernetes_executor.py

```diff
@@ -236,6 +273,103 @@ def test_worker_environment_dags_folder_using_git_sync(self):
+
+        worker_config = WorkerConfiguration(self.kube_config)
+        init_containers = worker_config._get_init_containers()
+
+        self.assertTrue(init_containers)  # check not empty
+        env = init_containers[0]['env']
+
+        self.assertTrue({'name': 'GIT_SSH_KEY_FILE', 'value': '/etc/git-secret/ssh'} in env)
+        self.assertTrue({'name': 'GIT_KNOWN_HOSTS', 'value': 'true'} in env)
+        self.assertTrue({'name': 'GIT_SSH_KNOWN_HOSTS_FILE',
+                        'value': '/etc/git-secret/known_hosts'} in env)
+        self.assertTrue({'name': 'GIT_SYNC_SSH', 'value': 'true'} in env)
+
+    def test_make_pod_git_sync_ssh_without_known_hosts(self):
+        # Tests the pod created with git-sync SSH authentication option is correct without known hosts
+        self.kube_config.airflow_configmap = 'airflow-configmap'
+        self.kube_config.git_ssh_key_secret_name = 'airflow-secrets'
+        self.kube_config.dags_volume_claim = None
+        self.kube_config.dags_volume_host = None
+        self.kube_config.dags_in_image = None
+
+        worker_config = WorkerConfiguration(self.kube_config)
+        kube_executor_config = KubernetesExecutorConfig(annotations=[],
+                                                        volumes=[],
+                                                        volume_mounts=[])
+
+        pod = worker_config.make_pod("default", str(uuid.uuid4()), "test_pod_id", "test_dag_id",
+                                     "test_task_id", str(datetime.utcnow()), 1, "bash -c 'ls /'",
+                                     kube_executor_config)
+
+        init_containers = worker_config._get_init_containers()
+        git_ssh_key_file = next((x['value'] for x in init_containers[0]['env']
+                                if x['name'] == 'GIT_SSH_KEY_FILE'), None)
+        volume_mount_ssh_key = next((x['mountPath'] for x in init_containers[0]['volumeMounts']
+                                    if x['name'] == worker_config.git_sync_ssh_secret_volume_name),
+                                    None)
+        self.assertTrue(git_ssh_key_file)
+        self.assertTrue(volume_mount_ssh_key)
+        self.assertEqual({'fsGroup': 65533}, pod.security_context)
+        self.assertEqual(git_ssh_key_file,
+                         volume_mount_ssh_key,
+                         ('The location where the git ssh secret is mounted'
+                          ' needs to be the same as the GIT_SSH_KEY_FILE path'))
+
+    def test_make_pod_git_sync_ssh_with_known_hosts(self):
+        # Tests the pod created with git-sync SSH authentication option is correct with known hosts
+        self.kube_config.airflow_configmap = 'airflow-configmap'
+        self.kube_config.git_ssh_secret_name = 'airflow-secrets'
+        self.kube_config.dags_volume_claim = None
+        self.kube_config.dags_volume_host = None
+        self.kube_config.dags_in_image = None
+
+        worker_config = WorkerConfiguration(self.kube_config)
```

Comment:
A test that `worker_config.git_sync_ssh_known_hosts_volume_name` is set to something would be good - we are using it in a later test, but we are never checking that this is set to anything directly

Reply:
`git_sync_ssh_known_hosts_volume_name` is a property set in the Class at initialization, (it does not get it's value from config but rather a hardcoded value)

I can add the following test that checks that those hardcoded values are set but I'm not sure of the value

```python
    def test_worker_config_properties(self):
        worker_config = WorkerConfiguration(self.kube_config)

        self.assertTrue(worker_config.dags_volume_name)
        self.assertTrue(worker_config.logs_volume_name)
        self.assertTrue(worker_config.git_sync_ssh_secret_volume_name)
        self.assertTrue(worker_config.git_ssh_key_secret_key)
        self.assertTrue(worker_config.git_sync_ssh_known_hosts_volume_name)
        self.assertTrue(worker_config.git_ssh_known_hosts_configmap_key)
```

Reply:
Oh yes, I hadn't twigged that. This is fine then, (though perhaps changing it `worker_config.GIT_SYNC_SSH_KNOWN_HOSTS_VOLUME_NAME` etc. That's a bit ... shouty ... though. Probably leave it as is.)